### DISCUSSION TITLE
Passing parent object to child objects

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#2947] [General]
Hi,
I am looking at some old code and found we have used 
 ```cs
   FieldManager.UpdateChildren(this);
```
We only need the primary key 'id' in child objects, so could we change the code to 
   ```cs
   FieldManager.UpdateChildren(id);
```

Kind regards

**Comment by rockfordlhotka** [comment]
This is a choice you make based on abstraction and dependency management.

Yes, you can pass in just the id, making the choice that the parent knows what the child needs. And if that changes in  the future, then you change the calling code.

Or you pass in the parent reference, making the choice that the parent doesn't know what the child needs, and allowing the child to pull whatever data is necessary.

I don't know that either is "correct" - it is a choice.

**Reply by michaelcsikos** [reply]
Sometimes in our generic base classes we declare `Insert` / `Update(object parent)` (or `ISomething parent`) for the same reason: we don't know what the child needs, and also the child class may have multiple types of parents.
